### DISCUSSION TITLE
SCF problem for DEEPKS model

### DISCUSSION THREAD

**Discussion by XiankeYue** [deepmodeling/deepks-kit#38] [Q&A] [answered]
Hi, I want to train a model using the DeepKS method. 
In the example of water_cluster, the model. pth is obtained by the DEEPKS method training, and then the Water Hexamer ring is self-consistent calculated by the DEEPKS model to obtain the energy and force information. I would like to know, in addition to the energy and force information, Can we get information about the energy level of the electron? Can we get a better result using DEEPKs.

**Comment by y1xiaoc** [comment, marked as answer]
Thanks for the question. It is possible to calculate energy level using deepks with pyscf. You can just check the pyscf doc and use the model manually with `deepks.scf.DSCF` class. However, since the model is only learned with energy and force, and since DFT methods are based on density and the KS orbitals are not physical, one should not expect to get better results in energy level. It is possible to train with the HOMO LUMO gap though. I guess @ouqi0711 should be the expert to this problem.

**Reply by XiankeYue** [reply]
Thank you for your reply. What should I do if I build the dataset that deepKS can use?  I just started machine learning, and I don't know much about it.  Is it feasible to do molecular dynamics for the isolated system to be calculated, to do a high-precision SCF calculation for each step of the structure, and to extract system coordinates, energy and force data? Is there a recommended computational software to build the input database? Thanks again.
